### PR TITLE
simplify by removing TagNames/TagName

### PR DESCRIPTION
**Pull request by grosser**
Did not really do much, but added a lot of ceremony ... maybe this logic is simpler/better ?
... definitely a lot cheaper (speed + GC wise) and easier for me to follow :)

... also added missing test for query normalization

### CODE REVIEW

File: lib/gutentag.rb [hunk 1]
```diff
@@ -10,7 +10,7 @@ def self.dirtier=(dirtier)
   end
 
   def self.normaliser
-    @normaliser ||= Gutentag::TagName
+    @normaliser ||= -> (t){ t.to_s.downcase }
```
Comment (pat) [inline lib/gutentag.rb]: Can this be changed to the old-school lambda syntax? Also, would rather avoid single-character parameters, so `t` should become `tag`.

File: lib/gutentag/tag_names.rb [hunk 2]
```diff
@@ -1,21 +0,0 @@
...
-  def self.call(*names)
-    new(names).call
-  end
-
-  def initialize(names)
-    @names = names.flatten
-  end
-
-  def call
-    name_values.collect { |name| Gutentag::TagName.call name }
```
Comment (pat) [inline lib/gutentag/tag_names.rb]: Can we keep TagNames, but have it use `Gutentag.normaliser` here? (Which it really should have been doing originally.)

File: spec/gutentag/active_record_spec.rb [hunk 3]
```diff
@@ -41,6 +42,14 @@
       it { is_expected.not_to include oregon_article }
     end
 
+    context 'given a denormalized tag name' do
```
Comment (pat) [inline spec/gutentag/active_record_spec.rb]: One thing I missed in the previous PR - Melbourne has a U in it, so you'll want to adjust tags/variable names accordingly :)

File: lib/gutentag.rb (hunk 1 above)
Comment (grosser) [inline lib/gutentag.rb]: should run the tests on 1.9 too if we want to keep compatibility ...used to writing this so I did not even notice ...

File: lib/gutentag/tag_names.rb (hunk 2 above)
Comment (grosser) [inline lib/gutentag/tag_names.rb]: why keep the extra indirection ?

Comment (grosser) [inline lib/gutentag/tag_names.rb]: ah ... I think I get what you had in mind ... make TagNames be 
```
tags.map { |t| Gutentag.normaliser.call(t.is_a?(Gutentag::Tag) ? t.name : t) }
```
... but that pattern does not make sense to me ... just a random method ... no need to wrap it into a class ...
if we followed that pattern then every method should be a class ... and that would be madness ...

File: spec/gutentag/active_record_spec.rb (hunk 3 above)
Comment (grosser) [inline spec/gutentag/active_record_spec.rb]: that's something best done on master or in a separate PR ... otherwise this will be a big/unreadable diff and cause merge conflicts ...

File: lib/gutentag/tag_names.rb (hunk 2 above)
Comment (pat) [inline lib/gutentag/tag_names.rb]: I was thinking TagNames was used in multiple places, hence making it re-usable. But you're right, it's not, so I guess we can lose it 👍

File: lib/gutentag/tagged_with_query.rb [hunk 4]
```diff
@@ -29,8 +29,9 @@ def id_query
   end
 
   def name_query
+    tag_names = tags.map { |t| Gutentag.normaliser.call(t.is_a?(Gutentag::Tag) ? t.name : t) }
```
Comment (pat) [inline lib/gutentag/tagged_with_query.rb]: Let's pull this line out into a private method (and switch the `t` variable to be `tag`)

File: spec/gutentag/active_record_spec.rb (hunk 3 above)
Comment (pat) [inline spec/gutentag/active_record_spec.rb]: Agreed. Let's get this merged in first, and then we can sort that out.

**Comment by grosser** [comment]
updated
